LocalNotifications/Sources/LocalNotifications/LocalNotifications.swift

```diff
@@ -0,0 +1,136 @@
+//
+// This code is public domain. You are free to use under the condition
+// you accept that no warranty is given or implied. Use at your own risk.
+//
+import SwiftGodot
+import Foundation
+import UserNotifications
+
+#initSwiftExtension(cdecl: "swift_entry_point", types: [LocalNotifications.self])
+
+@Godot
+class LocalNotifications:RefCounted {
+
+    let HOUR:Int = 60 * 60
+
+    var permitted = false
+    var notificationAccepted = false
+
+    /**
+           Display the local notification permission poup box. Check the value of `permitted` after completed.
+     */
+    @Callable
+    func requestPermission(onComplete:Callable) {
+        Task {
+            let notificationCenter = UNUserNotificationCenter.current()
+            let authorizationOptions: UNAuthorizationOptions = [.alert, .sound, .badge]
+        
+            let options = await notificationCenter.notificationSettings()
+            if options.badgeSetting == .enabled {
+                GD.print("Already Authorised")
+                self.permitted = true
+                onComplete.callDeferred()
+                return
+            }
+
+            do {
+                GD.print("Requesting Authorisation")
+                let authorizationGranted = try await notificationCenter.requestAuthorization(options: authorizationOptions)
+                self.permitted = authorizationGranted
+                onComplete.callDeferred()
+            } catch {
+                GD.print("Authorisation request failed. \(error.localizedDescription)")
+                self.permitted = false
+            }
+        }
+    }
+    
+    @Callable
+    func isPermitted() -> Bool {
+        self.permitted
+    }
+
+    @Callable
+    func wasNotificationAccepted() -> Bool {
+        self.notificationAccepted
+    }
+
+    @Callable
+    func removeAllNotifications() {
+        let notificationCenter = UNUserNotificationCenter.current()
+        let pending = notificationCenter.removeAllPendingNotificationRequests()
+    }
+
+    @Callable
+    func removeNotification(uid:String) {
+        let notificationCenter = UNUserNotificationCenter.current()
+        notificationCenter.removePendingNotificationRequests(withIdentifiers: [uid])
+    }
+
+    /**
+        Request that a notification banner be displayed in the future, as long
+        as the user is not inside the application. `reqeustPermission()` must
+        be called first.
+
+        - parameter uid: Random string or identifier used to delete this notification later if needed
+        - parameter title: title of the local notification.
+        - parameter body: text body of the local notification.
+        - parameter seconds: display notification in how many seconds time.
+        - returns no value is instantly expected. onComplete read the variable `notificationAccepted`
+     */
+    @Callable
+    func schedule(_ uid:String, _ title:String, _ body:String, _ seconds:Int, _ onComplete:Callable) {
```

Comment:
I don't think we should use underscores for the parameters in @Callable functions, it makes no difference for godot and if called from within swift it's better to have either proper names or just use the variable name, I generally try to replicate what the API calls them if available